### ISSUE TITLE
Add tests for logits processor correctness

### ISSUE THREAD

**Issue by maxdebayser** [vllm-project/vllm-spyre#295]
PR https://github.com/vllm-project/vllm-spyre/pull/290 updated our spyre input batch to use the new LogitsProcessor programming model.

We need to make sure not only that all sampling parameters that we support work for batch size 1, but also that they continue working as requests enter and leave the batch. For example, if we have a batch with multiple requests and one of them has a min_p setting, if another request leaves the batch, changing the size of the current batch, the dense indices are adjusted correctly so that min_p is not accidentally applied to another request.
